satpy/readers/ici_l1b_nc.py

```diff
@@ -0,0 +1,441 @@
+
+    @cached_property
+    def longitude_and_latitude(self):
+        """Get longitude and latitude coordinates."""
+        try:
+            return self._perform_geo_interpolation(
+                self[self.filetype_info['cached_longitude']],
+                self[self.filetype_info['cached_latitude']],
+                self._n_samples,
+            )
+        except KeyError:
+            logger.warning("Cached longitude and/or latitude datasets are not correctly defined in YAML file")  # noqa: E501
+        return None, None
+
+    @property
+    def start_time(self):
+        """Get observation start time."""
+        try:
+            start_time = datetime.strptime(
+                self['/attr/sensing_start_time_utc'],
+                '%Y%m%d%H%M%S.%f',
+            )
+        except ValueError:
+            start_time = datetime.strptime(
+                self['/attr/sensing_start_time_utc'],
+                '%Y-%m-%d %H:%M:%S.%f',
+            )
+        return start_time
+
+    @property
+    def end_time(self):
+        """Get observation end time."""
+        try:
+            end_time = datetime.strptime(
+                self['/attr/sensing_end_time_utc'],
+                '%Y%m%d%H%M%S.%f',
+            )
+        except ValueError:
+            end_time = datetime.strptime(
+                self['/attr/sensing_end_time_utc'],
+                '%Y-%m-%d %H:%M:%S.%f',
+            )
+        return end_time
+
+    @property
+    def spacecraft_name(self):
+        """Return spacecraft name."""
+        return self['/attr/spacecraft']
+
+    @property
+    def sensor(self):
+        """Return sensor."""
+        return self['/attr/instrument']
+
+    @property
+    def ssp_lon(self):
+        """Return subsatellite point longitude."""
+        # This parameter is not applicable to ICI?
+        return None
+
+    @property
+    def cached_observation_azimuth(self):
+        """Get observation azimuth angles."""
+        observation_azimuth, _ = self.observation_azimuth_and_zenith
+        return observation_azimuth
+
+    @property
+    def cached_observation_zenith(self):
+        """Get observation zenith angles."""
+        _, observation_zenith = self.observation_azimuth_and_zenith
+        return observation_zenith
+
+    @property
+    def cached_solar_azimuth(self):
+        """Get solar azimuth angles."""
+        solar_azimuth, _ = self.solar_azimuth_and_zenith
+        return solar_azimuth
+
+    @property
+    def cached_solar_zenith(self):
+        """Get solar zenith angles."""
+        _, solar_zenith = self.solar_azimuth_and_zenith
+        return solar_zenith
+
+    @property
+    def cached_longitude(self):
+        """Get longitude coordinates."""
+        longitude, _ = self.longitude_and_latitude
+        return longitude
+
+    @property
+    def cached_latitude(self):
+        """Get latitude coordinates."""
+        _, latitude = self.longitude_and_latitude
+        return latitude
```

Comment:
Why are these all called `cached_something`? I can't see any caching happening...

Reply:
That was bad naming so I removed the cache part from the name (it was the underlying functions that was using cache)